### PR TITLE
[JN-1243] Calculated value support

### PR DESCRIPTION
**Pull request by connorlbark**
#### DESCRIPTION (include screenshots, and mobile screenshots for participant UX)

Adds calculated values to the designer tool and the ability to preview and test them out within the UI.

image

It's a bit jittery while you type. I tried to figure out what update/memo was causing it, but I figured after not being able to fix it it's good enough for V1.

Based on feedback, I renamed the tab to `Derived Values`. I'm not sure if it's more or less helpful to rename code values as well since they're going to be called "calculated values" by surveyjs anyway.

#### TO TEST:  *(simple manual steps for confirming core behavior -- used for pre-release checks)*

- Go to any survey
- Add a  calculated value
- Edit it and observe the preview / try out the preview

### CODE REVIEW

**Comment by connorlbark** [comment]
converting back to draft to fix bugs

File: ui-admin/src/forms/designer/SplitCalculatedValueDesigner.tsx [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+      ))}
+    </div>
+  </>
+}
+
+const renderNewElementButton = (
+  formContent: FormContent,
+  onChange: (newContent: FormContent) => void,
+  calculatedValueIdx: number) => {
+  return <div className="my-2">
```
Comment (devonbush) [inline ui-admin/src/forms/designer/SplitCalculatedValueDesigner.tsx]: will be clearer if this function just takes responsibility for rendering the button, not a surrounding div.  You can add "py-2" to the classes of the surrounding div above to get the effect you need

File: ui-admin/src/forms/designer/SplitCalculatedValueDesigner.tsx [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+      <div className="d-flex">
+        {renderNewElementButton(content, onChange, -1)}
+      </div>
+      {content && content.calculatedValues && content.calculatedValues.map((_, elementIndex) => (
+        <div key={elementIndex} className="container">
+          <SplitCalculatedValueEditor
+            calculatedValueIndex={elementIndex}
+            editedContent={content}
+            onChange={onChange}/>
+          <div className="d-flex">
```
Comment (devonbush) [inline ui-admin/src/forms/designer/SplitCalculatedValueDesigner.tsx]: why does this need d-flex?

File: ui-admin/src/util/zendeskUtils.tsx [hunk 3]
```diff
@@ -18,7 +19,8 @@ const ZENDESK_PAGES: Record<DocsKey, string> = {
   SURVEY_EDIT: 'https://broad-juniper.zendesk.com/hc/en-us/articles/19885357297691-How-do-I-create-and-edit-surveys',
   SEARCH_EXPRESSIONS: 'https://broad-juniper.zendesk.com/hc/en-us/articles/26203593508251-Enrollee-Search-Expressions',
   PROXY_ENROLLMENT: 'https://broad-juniper.zendesk.com/hc/en-us/articles/26099797590555-Proxy-Enrollment',
-  WITHDRAWAL: 'https://broad-juniper.zendesk.com/hc/en-us/articles/20172265139995-How-to-withdraw-an-enrollee'
+  WITHDRAWAL: 'https://broad-juniper.zendesk.com/hc/en-us/articles/20172265139995-How-to-withdraw-an-enrollee',
+  DERIVED_VALUES: 'https://broad-juniper.zendesk.com/hc/en-us/articles/29123290381723-Derived-Values'
```
Comment (devonbush) [inline ui-admin/src/util/zendeskUtils.tsx:23]: don't forget to write the article!

**Review by devonbush — APPROVED** [review]
nice, a big improvement over the past!  worked great locally, I didn't notice any jerkiness while typing.  Some minor thoughts, and a reminder to write the zendesk article

**Review by MatthewBemis — APPROVED** [review]
Looks great. I also didn't notice any jitter while testing it out!

File: ui-admin/src/forms/designer/SplitCalculatedValueDesigner.tsx (hunk 2 above)
Comment (connorlbark) [reply ui-admin/src/forms/designer/SplitCalculatedValueDesigner.tsx]: spoiler alert: it doesn't 😄

**Pull request merged**